packages/lib/src/core/Services/fetch-json-data.ts

```diff
@@ -0,0 +1,66 @@
+import { FALLBACK_CONTEXT } from '../config';
+
+interface FetchJsonDataOptions {
+    accept?: string;
+    contentType?: string;
+    clientKey?: string;
+    errorMessage?: string;
+    headers?;
+    loadingContext?: string;
+    method?: string;
+    path: string;
+    errorLevel?: 'silent' | 'info' | 'warn' | 'error' | 'fatal';
+}
+
+function fetchJsonData(options: FetchJsonDataOptions, data?) {
+    const { clientKey, errorLevel = 'warn', loadingContext = FALLBACK_CONTEXT, path } = options;
+    const method = options.method || (data ? 'POST' : 'GET');
+
+    const request = {
+        method,
+        mode: 'cors',
+        cache: 'default',
+        credentials: 'same-origin',
+        headers: {
+            Accept: 'application/json, text/plain, */*',
+            'Content-Type': method === 'POST' ? 'application/json' : 'text/plain',
+            // NOTE: Remove next line to test old flow
+            ...(clientKey && { 'X-Client-Key': clientKey }),
+            ...options.headers
+        },
+        redirect: 'follow',
+        referrerPolicy: 'no-referrer-when-downgrade',
+        ...(data && { body: JSON.stringify(data) })
+    } as RequestInit;
+
+    // TODO: Delete
+    // const url = `${loadingContext}${path}${clientKey ? `?token=${clientKey}` : ''}`;
```

Comment:
Is this intentionally left here?

Reply:
Yes, the idea is that these changes can be tested before the `x-api-key` header is supported.
I'll delete it before merging the branch.